great_expectations/datasource/fluent/file_path_data_asset.py

```diff
@@ -110,7 +143,7 @@ class _FilePathDataAsset(DataAsset):
     _all_group_names: List[str] = pydantic.PrivateAttr()
 
     # `_data_connector`` should be set inside `_build_data_connector()`
-    _data_connector: DataConnector = pydantic.PrivateAttr()
+    _data_connector: FilePathDataConnector = pydantic.PrivateAttr()
```

Comment:
as far as I can tell every valid DataConnector for this asset must inherit from FilePathDataConnector.